src/timeline.ts

```diff
@@ -251,43 +254,72 @@ class Timeline implements IAppView {
       }
     }
 
+    // helper variable for on click event
+     let open2dHistogram = null;
 
 
-    // helper variable for on click event
-    let open2dHistogram = null;
+   // creating 2D Ratio bars
 
-    // creating 2D Ratio bars
     function generateBars(width) {
       return idPairs.map((pair) => {
-        console.log('start loading pair', pair);
-
-        return Promise.all([ajax.getAPIJSON(`/taco/diff_log/${pair[0]}/${pair[1]}/1/1/2/structure,content`), pair])
-          .then((args) => {
+      return Promise.all([ajax.getAPIJSON(`/taco/diff_log/${pair[0]}/${pair[1]}/1/1/2/structure,content`), pair])
+         .then((args) => {
             const json = args[0];
             const pair = args[1];
 
             const pairPosX = pair.map((d) => parseFloat(d3.select(`#circle_${d}`).attr('cx')));
 
-            console.log('finished loading pair - BARS', pair, pairPosX, json);
+           // console.log('finished loading pair - BARS', pair, pairPosX, json);
 
             const w = 80;
             const h = 30;
             const barPadding = 0.5;
 
             const data = [json.no_ratio, json.a_ratio, json.c_ratio, json.d_ratio];
 
-            const color = d3.scale.ordinal()
-              .domain(<any>[0, data.length - 1])
-              .range(['#D8D8D8', '#67C4A7', '#8DA1CD', '#F08E65']);
+            const data_removed = [json.no_ratio, json.a_ratio, json.c_ratio];
+
+            const data_content = [json.no_ratio, json.a_ratio, json.d_ratio];
+
+           const data_added = [json.no_ratio, json.c_ratio, json.d_ratio];
+
+           const data_nochange = [json.a_ratio, json.c_ratio, json.d_ratio];
+
+           console.log(data, json);
+
+
+           const color = d3.scale.ordinal()
+              /*.domain(<any>[0, data.length-1])*/
+              .range(['#D8D8D8', '#67c4a7', '#8DA1CD', '#F08E65']);
```

Comment:
What's the reason for using different color scales per change type? Why not using a simple color?